WordPress/src/main/java/org/wordpress/android/ui/uploads/UploadService.java

```diff
@@ -407,6 +429,18 @@ private PostModel removeQueuedPostByLocalId(int localPostId) {
         return null;
     }
 
+    private void addFailedReasonToFailedPosts(PostModel post, FailReason reason) {
+        sFailedUploadPosts.put(post.getId(), reason);
+    }
+
+    public static boolean removeFailedReasonForPost(PostModel post) {
+        return (sFailedUploadPosts.remove(post.getId()) != null ? true : false);
```

Comment:
We can simplify this and squash a lint warning by using `return (sFailedUploadPosts.remove(post.getId()) != null);`.